### DISCUSSION TITLE
Run timeline based on condition

### DISCUSSION THREAD

**Discussion by FelicityF** [jspsych/jsPsych#1121] [Q&A] [answered]
Hi! 

I have 5 different feedback conditions and I would like to run one of the versions based on the version number that is identified at the start of the experiment. The variable condition is empty at the start and will be filled by the selection of the participant:

`var condition = []

[...]

var conditions = {
  type: 'survey-multi-choice',
  questions: [{
    prompt: "Please select assigned version number",
    options: ["1", "2", "3", "4", "5"],
    required: true
  }, ],
  on_finish: function(data) {
    var responses = JSON.parse(data.responses);
    var code = responses.Q0;
    jsPsych.data.addProperties({
      conditions: code
    });
    condition[0] = jsPsych.data.get().last(1).values()[0].conditions}
};

timeline.push(Subject_ID, age, gender, language, conditions)`

This all seems to work fine. However, based on that value, I want to run a particular timeline. Because the type of feedback is defined within this timeline. But how I have it now, the experiment skips over the actual experiment and goes straight from the instructions to the "this is the end of the experiment" text.

`for (var i = 0; condition.length; i++){
    if (condition[i] == "1"){
    timeline.push(procedure_training1NO, pauze, procedure_training2NO, procedure_testNO, procedure_verbtest, procedure_nountest)
} else if (condition[i] == "2")
 {timeline.push(procedure_training1NEG75, pauze, procedure_training2NEG75, procedure_testNEG75, procedure_verbtest, procedure_nountest)
} else if (condition[i] == "3") 
{timeline.push(procedure_training1NEG50, pauze, procedure_training2NEG50, procedure_testNEG50, procedure_verbtest, procedure_nountest)
} else if (condition[i] == "4")
 {timeline.push(procedure_training1POS75, pauze, procedure_training2POS75, procedure_testPOS75, procedure_verbtest, procedure_nountest)
} else if (condition[i] == "5") 
{timeline.push(procedure_training1POS50, pauze, procedure_training2POS50, procedure_testPOS50, procedure_verbtest, procedure_nountest)
}
};
`

Any idea how to solve this? Thank you!

**Comment by becky-gilbert** [comment, marked as answer]
Hi @FelicityF! I think the problem with this approach is that your `for` loop is running when the page first loads, which means that your `condition` array is still empty, so nothing is ever added to the timeline. Instead, you want the code in your `for` loop to run _during_ your experiment, after the participant has entered their response in the `conditions` trial. 

One way to fix this would be to use the [jsPsych.addNodeToEndOfTimeline function](https://www.jspsych.org/core_library/jspsych-core/#jspsychaddnodetoendoftimeline) (see examples in the docs) inside of the `conditions` trial's `on_finish` function to add new timelines after this trial. The specifics of the new timelines that are added would depend on the value of `code`. If you do it this way then you won't need the `condition` array, and you'll only put the `conditions` trial on your main experiment `timeline` (in `jsPsych.init`). You'll also want to add your "this is the end of the experiment" trial to the end of your whole experiment using `jsPsych.addNodeToEndOfTimeline`, after adding any other timelines in the `conditions` trial `on_finish` function. Does that make sense? 

Note that there's currently a [bug in the addNodeToEndOfTimeline function](https://github.com/jspsych/jsPsych/issues/688) that means that the callback argument isn't optional (we'll fix this in the next release). Tn the meantime you can use an empty function as the callback argument, like this: 
```javascript
jsPsych.addNodeToEndOfTimeline(new_timeline, function() {});
```

Lastly, see [this discussion post](https://github.com/jspsych/jsPsych/discussions/1113) about how to add a nicely-formatted code block to your question 😃
